autoprecompiles/src/pgo/cell/mod.rs

```diff
@@ -88,69 +97,107 @@ impl<A: Adapter + Send + Sync, C: Candidate<A> + Send + Sync> PgoAdapter for Cel
 
     fn create_apcs_with_pgo(
         &self,
-        mut blocks: Vec<BasicBlock<<Self::Adapter as Adapter>::Instruction>>,
+        blocks: Vec<Block<<Self::Adapter as Adapter>::Instruction>>,
+        block_exec_count: Option<HashMap<usize, u32>>,
```

Comment:
Could this be `Option<Vec<_>>`? Also does the Vec always have the same length as the `blocks` Vec above?